docusaurus/docs/introduction/plugin-lifecycle.md

```diff
@@ -0,0 +1,89 @@
+
+:::
+
+## Loading plugins
+
+Plugins are loaded either when Grafana starts up or when a plugin has been installed/uninstalled during runtime. 
+
+Understanding the different phases involved when Grafana is loading a plugin may help you better understand plugin usage and troubleshoot any unexpected behavior, such as why a certain plugin is not marked as installed in the plugins catalog and/or for use within Grafana even though you've installed it. You can check the [Grafana server log](https://grafana.com/docs/grafana/latest/troubleshooting/#troubleshoot-with-logs) for any unexpected errors or details related to loading a plugin. In addition, you can enable even more details by changing the [log level to debug](https://grafana.com/docs/grafana/latest/setup-grafana/configure-grafana/#log).
+
+For [backend](./backend.md) plugins, there is an additional initialization process (see [Phase 4](#phase-4-backend-plugin-initialization)).
+
+### Phase 1. Plugin discovery 
+
+Grafana starts to discover which plugins are installed by scanning directories on the file system for every `plugin.json`.
+
+### Phase 2. Plugin loading 
+
+All plugins that were discovered in the discovery phase are checked to make sure they’re valid. Some of the automated checks include:
+- Plugins must have a valid [signature](https://grafana.com/docs/grafana/latest/administration/plugin-management/#plugin-signatures). Valid plugins are referred to as _verified plugins_. 
+- Angular detection: Given that [Angular is deprecated](https://grafana.com/docs/grafana/latest/developers/angular_deprecation/), if Angular support is disabled and Angular is detected in the plugin, then we record an error and don't allow the plugin to be loaded.
+
+### Phase 3. Backend plugin initialization
+
+For any verified plugin that has a backend, Grafana configures the backend client to use HashiCorp’s Go Plugin System over RPC.
+
+### Phase 4. Registration
+
+All verified plugins are registered in an in-memory registry. From now on, the plugin is available within Grafana and so are referred to as _registered plugins_.
+
+Registered plugins show as installed in the catalog and appear in views for selecting a panel or data source in a dashboard.
+
+### Phase 5. Start backend plugin
+
+For registered plugins that have a backend, Grafana starts to run the backend binary as a separate process using HashiCorp’s Go Plugin System over RPC. The supported plugin protocol and version is negotiated between Grafana (client) and the plugin (server) to give Grafana an understanding of the plugin's capabilities. 
+
+A Grafana backend plugin has its own separate life cycle. So long as the backend plugin is running, Grafana will make sure to restart the backend plugin in case it crashes or is terminated. When Grafana is shut down, the backend processes is then terminated.  
```

Comment:
> A Grafana backend plugin has its own separate life cycle

I'm not sure this statement is clear enough, separate to what?

> So long as the backend plugin is running, Grafana will make sure to restart the backend plugin in case it crashes or is terminated.

I find this a little unclear, if the backend plugin crashes, it is no longer running and needs to be restarted, so the opening statement feels confusing.

Reply:
@marefr - again, could use your expertise to clarify these points